### PR TITLE
Dynamic launch functionality

### PR DESCRIPTION
**Pull request by yourFavoriteSoftwareDeveloper**

### CODE REVIEW

File: app/src/main/java/com/dankideacentral/dic/LaunchActivity.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        super.onCreate(savedInstanceState);
+        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
+        String twitterAuth = preferences.getString(getString(R.string.twitter_auth_preference), "");
+        String fineLocationPermission = "android.permission.ACCESS_FINE_LOCATION";
+        PackageManager pm = this.getPackageManager();
+        int hasFineLocationPermission = pm.checkPermission(
+                fineLocationPermission,
+                this.getPackageName());
+        // auth tokens don't expire. Just check if it exists
+        if ("".equals(twitterAuth)) {
```
Comment (ermel272) [inline app/src/main/java/com/dankideacentral/dic/LaunchActivity.java]: Only suggestion I really have is maybe to make the "" a final static class constant

**Review by ermel272 — APPROVED** [review]
Solid man, only comment I really had was with the hard coded empty string

File: app/src/main/java/com/dankideacentral/dic/LaunchActivity.java (hunk 1 above)
Comment (yourFavoriteSoftwareDeveloper) [reply app/src/main/java/com/dankideacentral/dic/LaunchActivity.java]: So create an "empty string" Constant? I was only doing it this way so I didn't have to null check twitterAuth.

Comment (yourFavoriteSoftwareDeveloper) [reply app/src/main/java/com/dankideacentral/dic/LaunchActivity.java]: okay i got it...

Comment (yourFavoriteSoftwareDeveloper) [reply app/src/main/java/com/dankideacentral/dic/LaunchActivity.java]: made the change

**Review by srowhani — COMMENTED** [review]
Sweet. Looks good :+1:

**Pull request merged**